File: MiniDOMOffice/Magazzino/Inventario/Collegamenti/OggettoCollegatoCursor.vb
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Public Class Office

    <Serializable>
    Public Class OggettoCollegatoCursor
        Inherits DBObjectCursor(Of OggettoCollegato)

        Private m_DataCollegamento As New CCursorField(Of Date)("DataCollegamento")
        Private m_DataScollegamento As New CCursorField(Of Date)("DataScollegamento")
        Private m_IDOggetto1 As New CCursorField(Of Integer)("IDOggetto1")
        Private m_IDOggetto2 As New CCursorField(Of Integer)("IDOggetto2")
        Private m_Relazione As New CCursorField(Of RelazioniOggettiCollegati)("Relazione")

        Public Sub New()
        End Sub

        Public ReadOnly Property DataCollegamento As CCursorField(Of Date)
            Get
                Return Me.m_DataCollegamento
            End Get
        End Property

        Public ReadOnly Property DataScollegamento As CCursorField(Of Date)
            Get
                Return Me.m_DataScollegamento
            End Get
        End Property

        Public ReadOnly Property IDOggetto1 As CCursorField(Of Integer)
            Get
                Return Me.m_IDOggetto1
            End Get
        End Property

        Public ReadOnly Property IDOggetto2 As CCursorField(Of Integer)
            Get
                Return Me.m_IDOggetto2
            End Get
        End Property

        Public ReadOnly Property Relazione As CCursorField(Of RelazioniOggettiCollegati)
            Get
                Return Me.m_Relazione
            End Get
        End Property

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Nothing
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeOggettiCollegati"
        End Function


    End Class

End Class

File: MiniDOMOffice/Magazzino/Inventario/Collegamenti/OggettoCollegatoCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Store;


namespace minidom
{
    public partial class Store
    {
        /// <summary>
        /// Cursore di oggetti <see cref="OggettoCollegato"/>
        /// </summary>
        [Serializable]
        public class OggettoCollegatoCursor 
            : Databases.DBObjectCursor<OggettoCollegato>
        {
            private DBCursorField<DateTime> m_DataCollegamento = new DBCursorField<DateTime>("DataCollegamento");
            private DBCursorField<DateTime> m_DataScollegamento = new DBCursorField<DateTime>("DataScollegamento");
            private DBCursorField<int> m_IDOggetto1 = new DBCursorField<int>("IDOggetto1");
            private DBCursorField<int> m_IDOggetto2 = new DBCursorField<int>("IDOggetto2");
            private DBCursorField<RelazioniOggettiCollegati> m_Relazione = new DBCursorField<RelazioniOggettiCollegati>("Relazione");

            /// <summary>
            /// Costruttore
            /// </summary>
            public OggettoCollegatoCursor()
            {
            }

            /// <summary>
            /// DataCollegamento
            /// </summary>
            public DBCursorField<DateTime> DataCollegamento
            {
                get
                {
                    return m_DataCollegamento;
                }
            }

            /// <summary>
            /// DataScollegamento
            /// </summary>
            public DBCursorField<DateTime> DataScollegamento
            {
                get
                {
                    return m_DataScollegamento;
                }
            }

            /// <summary>
            /// IDOggetto1
            /// </summary>
            public DBCursorField<int> IDOggetto1
            {
                get
                {
                    return m_IDOggetto1;
                }
            }

            /// <summary>
            /// IDOggetto2
            /// </summary>
            public DBCursorField<int> IDOggetto2
            {
                get
                {
                    return m_IDOggetto2;
                }
            }

            /// <summary>
            /// Relazione
            /// </summary>
            public DBCursorField<RelazioniOggettiCollegati> Relazione
            {
                get
                {
                    return m_Relazione;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Store.OggettiInventariati.OggettiCollegati;
            }

            //public override string GetTableName()
            //{
            //    return "tbl_OfficeOggettiCollegati";
            //}
        }
    }
}